lib/matplotlib/collections.py

```diff
@@ -299,6 +299,7 @@ def draw(self, renderer):
         gc = renderer.new_gc()
         self._set_gc_clip(gc)
         gc.set_snap(self.get_snap())
+        gc.set_alpha(self._alpha)
```

Comment:
It is a bigger change, but I think that this means we can also remove a whole bunch of the complexity of managing the edge/face colors on collections.  If you look at `set_alpha` on collections you see we call `super().set_alpha()` and then call the private versions of `_set_facecolor` and `_set_edgecolor` with stashed versions of the original colors and then forces the alpha on them.  I _think_ that can go away.

Reply:
Well, stood out when looking at the collections code is that alpha is a second-class citizen, attribute-wise: you can give different face/edge colors to the collected elements, but the alpha always gets replaced by the collections' alpha. That made it easier to deem`gc.set_alpha(self._alpha)` safe. It does seem inconsistent, though. What should the goal be here? Simplicity or consistency?

Reply:
My suggestion is to make that a separate PR if it is deemed the correct thing to do. There is a lot of complexity there, some of which may be for a reason.